### PR TITLE
Replace all instances of hardcoded Colors with shared color resources

### PR DESCRIPTION
**Pull request by ThomasMPont**
I've replaced all hardcoded colors with shared colors from the colors.xml file. The only hardcoded color left in the codebase is `Color.Transparent` which I opted to leave.

This PR is a minor cleanup task. I still suggest you take a tour of the app to check if the colors look right. I've had to slightly shift a few unique colors that were being used.

### CODE REVIEW

**Comment by ThomasMPont** [comment]
It seems SonarCloud is calling some of my changes new duplications, despite the old code being just as repetitive. I suggest we still merge this, and look into duplication at a later date.

**Review by samymessouak — APPROVED** [review]
Nice rework.

**Comment by nada-ezziti** [comment]
Removing hardcoded values is great for improving the flexibility and maintainability of the application. Nice work.

**Review by nada-ezziti — APPROVED** [review]
Removing hardcoded values is great for improving the flexibility and maintainability of the application. Nice work.

**Comment by samymessouak** [comment]
To avoid the conflict : 
- Go on your branch 
- Run "git merge main"
- commit and push changes

**Review by samymessouak — DISMISSED** [review]
To avoid the conflict :

Go on your branch
Run "git merge main"
commit and push changes

**Review by SaFouNaldo — APPROVED** [review]
I don't really see the point of using R.color instead of Color, but for many classes it does not change the quality or readability of the code.

I marked it as approved for the appreciated effort

**Review by SaFouNaldo — DISMISSED** [review]
There are conflicts with branch main make sure to work on a "updated" branch

File: app/src/main/java/com/github/se/signify/ui/screens/LoginScreen.kt [hunk 1]
```diff
@@ -74,11 +74,11 @@ fun LoginScreen(navigationActions: NavigationActions) {
       Brush.verticalGradient(
           colors =
               listOf(
-                  Color(0xFFFFFFFF),
-                  Color(0xFFFFFFFF), // White at the top
-                  Color(0xFFFFFFFF),
-                  Color(0xFF05A9FB) // Light blue at the bottom
```
Comment (ziyadbagh) [inline app/src/main/java/com/github/se/signify/ui/screens/LoginScreen.kt:80]: Nice work, it will allow us to scale faster

**Review by ziyadbagh — APPROVED** [review]

**Pull request merged**